### PR TITLE
Fix HTTP/2 listener stop issue

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

Related to : `immediateStop does not work as expected for Http2 #3553`(https://github.com/ballerina-platform/ballerina-standard-library/issues/3553)

## Examples

```bal
import ballerina/io;
import ballerina/lang.runtime;
import ballerina/http;

public function main() returns error? {
    http:Listener 'listener = check new(9090);
    http:Service 'service = service object {
        resource function get hello () returns string {
            return "Hello ";
        }
    };
    check 'listener.attach('service);
    check 'listener.'start();

    http:Client client1 = check new("localhost:9090");
    string res = check client1->get("/hello");
    io:println(res);

    check 'listener.immediateStop();
    runtime:sleep(5);

    http:Client client2 = check new("localhost:9090");
    res = check client2->get("/hello");
    io:println(res);
}
```

Run the file using `bal run main.bal`
Output :
```
Hello
error: Something wrong with the connection
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] <s>Updated the spec</s>
- [ ] <s>Checked native-image compatibility</s>

### CODE REVIEW

File: ballerina-tests/tests/http2_listener_stop_tests.bal [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    response = client2->/hello;
+    if response is error {
+        test:assertEquals(response.message(), "Something wrong with the connection");
+    } else {
+        test:assertFail(msg = "Found unexpected output: " + response);
+    }
+}
+
+@test:Config {}
+public function testHttp2SecuredListenerStop() returns error? {
```
Comment (Bhashinee) [inline ballerina-tests/tests/http2_listener_stop_tests.bal:80]: Shall we add another test case with HTTPS + prior knowledge? It should go in the same path as in this test. So ideally it is enough to add only this. Just to be on the safe side and to cover all the possible scenarios?

Comment (TharmiganK) [reply ballerina-tests/tests/http2_listener_stop_tests.bal:80]: Sure, will add one 👍

Comment (TharmiganK) [reply ballerina-tests/tests/http2_listener_stop_tests.bal]: Fixed via [f0efef6](https://github.com/ballerina-platform/module-ballerina-http/pull/1344/commits/f0efef63950997879165bdc55d451b9c50209325)

File: ballerina-tests/tests/http2_listener_stop_tests.bal [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    check serviceEP.immediateStop();
+    runtime:sleep(5);
+
+    response = client1->/hello;
+    if response is error {
+        test:assertEquals(response.message(), "Something wrong with the connection");
+    } else {
+        test:assertFail(msg = "Found unexpected output: " + response);
```
Comment (Bhashinee) [inline ballerina-tests/tests/http2_listener_stop_tests.bal:34]: We can remove this `msg =`

**Review by Bhashinee — APPROVED** [review]

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**